var/spack/repos/builtin/packages/mpip/package.py

```diff
@@ -4,29 +4,121 @@
+
+    variant('maxargs', values=int, default=32,
+            description='Set number of command line arguments in report')
+
+    variant('stackdepth', values=int, default=8,
+            description='Specify maximum report stacktrace depth')
+
+    variant('internal-stackdepth', values=int, default=3,
+            description='Specify number of internal stack frames')
 
     variant('shared', default=False, description="Build the shared library")
-    variant('demangling', default=False, description="Build with demangling support")
-    variant('setjmp',
-            default=False,
-            description="Replace glic backtrace() with setjmp for stack trace")
 
-    depends_on("elf")
-    depends_on("libdwarf")
-    depends_on('libunwind', when=os.uname()[4] == "x86_64")
-    depends_on("mpi")
+    @when('@3.5:')
+    def configure_args(self):
+        spec = self.spec
+
+        config_args = []
+        config_args.append("--with-cc=%s" % spec['mpi'].mpicc)
+        config_args.append("--with-cxx=%s" % spec['mpi'].mpicxx)
+        config_args.append("--with-f77=%s" % spec['mpi'].mpif77)
+
+        #  mpiP checks for libiberty demangling during configure.
+        #  Current mpiP configure functionality allows specifying alternate
+        #  demangling support (IBM, Compaq) as an argument, but the usefulness
+        #  of this is not clear.
+        #
+        #  Since, --enable-demangling doesn't do anything,
+        #  providing --disable-demangling in the event that there is an error
+        #  with demangling.
+        if '-demangling' in spec:
+            config_args.append('--disable-demangling')
+
+        if '-mpi-io' in spec:
+            config_args.append('--disable-mpi-io')
+
+        if '-mpi-rma' in spec:
+            config_args.append('--disable-mpi-rma')
+
+        if '-mpi-nbc' in spec:
+            config_args.append('--disable-mpi-nbc')
+
+        if '-bfd' in spec:
+            config_args.append('--disable-bfd')
+
+        if '-libunwind' in spec:
+            config_args.append('--disable-libunwind')
+
+        #  Simply enabling setjmp may result in libunwind being used,
+        #  if available.  Adding --disable-libunwind to ensure setjmp is used.
+        if '+setjmp' in spec:
+            config_args.append('--disable-libunwind')
+            config_args.append('--enable-setjmp')
+
+        maxargs = int(spec.variants['maxargs'].value)
+        if maxargs != 32:
+            config_args.extend(['--enable-maxargs={0}'.format(maxargs)])
+
+        stackdepth = int(spec.variants['stackdepth'].value)
+        if stackdepth != 8:
+            config_args.extend(['--enable-stackdepth={0}'.format(stackdepth)])
+
+        internal_stackdepth = int(spec.variants['internal-stackdepth'].value)
+        if internal_stackdepth != 3:
+            config_args.extend(['--enable-internal-stackdepth={0}'
+                               .format(internal_stackdepth)])
```

Comment:
I think it's better to pass the default, since the default may change in the future.